File: ruby/Problem36.rb
# The decimal number, 585 = 10010010012 (binary), is palindromic in both bases.
# 
# Find the sum of all numbers, less than one million, which are palindromic in base 10 and base 2.
# 
# (Please note that the palindromic number, in either base, may not include leading zeros.)

def parindromic?(x)
  s = x.to_s
  s == s.reverse
end

def binary_parindromic?(x)
  s = x.to_s(2)
  s == s.reverse
end

p (0..1000000)
  .select(&method(:parindromic?))
  .select(&method(:binary_parindromic?))
  .sum

File: java/Problem36.java
public class Problem36 {
    private static class Palindromes {
        private Palindromes() {
        }

        public static boolean isPalindrome(String s) {
            StringBuilder builder = new StringBuilder(s);
            builder.reverse();
            return builder.toString().equals(s);
        }
    }

    public static void main(String[] args) {
        int sum = 0;
        for (int i = 0; i < 1000000; i++) {
            String decimal = Integer.toString(i);
            if (Palindromes.isPalindrome(decimal)) {
                String binary = Integer.toBinaryString(i);
                if (Palindromes.isPalindrome(binary)) {
                    sum += i;
                }
            }
        }
        System.out.println(sum);
    }
}
